index.js

```diff
@@ -174,6 +188,18 @@ class ClinicDoctor extends events.EventEmitter {
       })
     )
 
+    readStream(analysisStringified)
+      .then((result) => {
+        const issue = JSON.parse(result).issueCategory
```

Comment:
This is kinda taking the scenic route :)

The `result` value was just `JSON.stringify`-d when the `analysisStringified` stream was created a few lines up—we could assign `this.issue` in that `transform()` callback, or maybe a bit cleaner, we could pull the `new Analysis()` call into its own variable and attach an `on('data')` handler to it to assign the `this.issue` value there.

Reply:
Ah thanks, I didn't see that. That makes it much nicer